espree.js

```diff
@@ -4356,7 +4384,7 @@ function parseParam(options) {
         if (rest) {
             throwErrorTolerant(lookahead, Messages.DefaultRestParameter);
         }
-        if (!allowDefaultParams) {
+        if (!(allowDefaultParams || allowDestructuring)) {
```

Comment:
`if (!allowDefaultParams || !allowDesctructuring)` has a slightly diff meaning and won't work.

how about:

```js
if (allowDefaultParams || allowDestructuring) {
	lex();
	def = parseAssignmentExpression();
	++options.defaultCount;
} else {
	throwUnexpected(lookahead);
}
```